src/language_provider.ts

```diff
@@ -483,6 +485,49 @@ export default class PromQlLanguageProvider extends LanguageProvider {
     }
   }
 
+  getSeriesLabels = async (selector: string, otherLabels: Label[]): Promise<string[]> => {
+    let possibleLabelNames, data: Record<string, string[]>;
+
+    // Exclude __name__ from output
+    otherLabels.push({ name: '__name__', value: '', op: '!=' });
+    data = await this.fetchSeriesLabelsMatch(selector);
+    possibleLabelNames = Object.keys(data);
+
+    const usedLabelNames = new Set(otherLabels.map((l) => l.name)); // names used in the query
+    return possibleLabelNames.filter((l) => !usedLabelNames.has(l));
+  };
+
+  fetchSeriesLabelsMatch = async (name: string): Promise<Record<string, string[]>> => {
+    const interpolatedName = this.datasource.interpolateString(name);
+    const range = this.datasource.getAdjustedInterval(this.timeRange);
+    const urlParams = {
+      ...range,
+      'match[]': interpolatedName,
+    };
+    const url = `/api/v1/labels`;
+
+    const data: string[] = await this.request(url, [], urlParams);
+    // Convert string array to Record<string , []>
+    return data.reduce((ac, a) => ({ ...ac, [a]: '' }), {});
```

Comment:
from the comment i see that the `Record` should be `Record<string , []>`, but in the reduce the object is `{[string]: string}`. Maybe I miss something

Reply:
Yes, there's a mistake here. I checked, only keys are used everywhere, so it's more logical to leave it as `string[]`.